Assets/MixedRealityToolkit/MSBuild/Scripts/CSProjectInfo.cs

```diff
@@ -0,0 +1,363 @@
+
+                File.WriteAllLines(Path.Combine(propsOutputFolder, $"{Guid.ToString()}.csmap"), sourceGuidToClassName.Select(t => $"{t.Key}:{t.Value}"));
+
+                bool bothConfigurations = PlayerPlatforms.Count > 0;
+
+                List<string> supportedPlatformBuildConditions = new List<string>();
+                PopulateSupportedPlatformBuildConditions(supportedPlatformBuildConditions, suportedPlatformBuildConditionTemplate, "InEditor", InEditorPlatforms);
+                PopulateSupportedPlatformBuildConditions(supportedPlatformBuildConditions, suportedPlatformBuildConditionTemplate, "Player", PlayerPlatforms);
+
+                HashSet<string> inEditorSearchPaths = new HashSet<string>(), playerSearchPaths = new HashSet<string>();
+                string projectReferences = string.Join("\r\n", CreateProjectReferencesSet(projectReferenceSetTemplate, inEditorSearchPaths, true), CreateProjectReferencesSet(projectReferenceSetTemplate, playerSearchPaths, false));
+                Dictionary<string, string> tokens = new Dictionary<string, string>()
+                {
+                    { "<!--PROJECT_GUID_TOKEN-->", Guid.ToString() },
+                    { "<!--ALLOW_UNSAFE_TOKEN-->", Assembly.compilerOptions.AllowUnsafeCode.ToString() },
+                    { "<!--LANGUAGE_VERSION_TOKEN-->", MSBuildTools.CSharpVersion },
+
+                    { "<!--DEVELOPMENT_BUILD_TOKEN-->", "false" }, // Default to false
+
+                    { "<!--OUTPUT_PATH_TOKEN-->", Path.Combine("..", "MRTKBuild") },
+                    { "<!--IS_EDITOR_ONLY_TARGET_TOKEN-->", (ProjectType ==  ProjectType.EditorAsmDef || ProjectType == ProjectType.PredefinedEditorAssembly).ToString() },
+                    { "<!--UNITY_EDITOR_INSTALL_FOLDER-->", Path.GetDirectoryName(EditorApplication.applicationPath) + "\\"},
+
+                    { "<!--DEFAULT_PLATFORM_TOKEN-->", availablePlatforms.First(t=>t.BuildTarget == BuildTarget.StandaloneWindows).Name },
+
+                    { "<!--SUPPORTED_PLATFORMS_TOKEN-->", string.Join(";", availablePlatforms.Select(t=>t.Name)) },
+
+                    { "<!--INEDITOR_ASSEMBLY_SEARCH_PATHS_TOKEN-->", string.Join(";", inEditorSearchPaths) },
+                    { "<!--PLAYER_ASSEMBLY_SEARCH_PATHS_TOKEN-->", string.Join(";", playerSearchPaths) },
+
+                    { "##PLATFORM_PROPS_FOLDER_PATH_TOKEN##", propsOutputFolder },
+
+                    { projectReferenceSetTemplate, projectReferences },
+                    { sourceIncludeTemplate, string.Join("\r\n", sourceIncludes) },
+                    { suportedPlatformBuildConditionTemplate, string.Join("\r\n", supportedPlatformBuildConditions) }
+                };
+
+                projectFileTemplateText = Utilities.ReplaceTokens(projectFileTemplateText, tokens);
+            }
+            else
+            {
+                Debug.LogError("Failed to find ProjectReferenceSet and/or Source_Include templates in the project template file.");
```

Comment:
> Debug.LogError("Failed to find ProjectReferenceSet and/or Source_Include templates in the project template file."); [](start = 15, length = 116)

Nit: It might be helpful to log the missing files.